src/components/king-of-the-hill/LeaderboardItem.tsx

```diff
@@ -83,17 +84,26 @@ export const LeaderboardItem = memo(function LeaderboardItem({
       <View style={styles.itemContainer}>
         {/* icon */}
         <View style={styles.iconColumn}>
-          {iconUrl && (
-            <View style={styles.tokenIconShadow}>
-              <View style={styles.tokenIconContainer}>
+          <View style={styles.tokenIconShadow}>
+            <View style={styles.tokenIconContainer}>
+              {iconUrl ? (
                 <RainbowImage source={{ url: iconUrl }} style={styles.tokenIcon} />
-                {(ranking === 2 || ranking === 3) && <View style={[styles.rankOverlay, { borderColor: rankingStyle.backgroundColor }]} />}
-              </View>
-              <View style={styles.chainImageContainer}>
-                <ChainImage chainId={token.chainId} size={16} position="absolute" />
-              </View>
+              ) : (
+                <RainbowCoinIcon
+                  color={token.colors?.primary || token.colors?.fallback || undefined}
+                  chainId={token.chainId}
+                  symbol={token.symbol}
+                  showBadge={false}
+                />
+              )}
+              {iconUrl && (ranking === 2 || ranking === 3) && (
```

Comment:
we can still have the rank overlay even without the iconUrl now yeah? given that it falls back to the RainbowCoinIcon